inst/tutorials/grading-demo/grading-demo.Rmd

```diff
@@ -258,35 +254,90 @@ sqrt(log(2))
 grade_code("Don't worry, things will soon get harder.")
 ```
 
-* Now, throw an error message of `"boom"`
+## Grading exercise errors
 
-```{r grade_error, exercise = TRUE}
+Since `enable_gradethis()` sets `learnr::tutorial_options(exercise.error.checker = grade_learnr_error)`, when an exercise submission produces an evaluation error, intelligent feedback is provided via `grade_code()` (when a solution is provided). Try it yourself by submitting `mt` (R error, incorrect answer), then `cars` (No error, incorrect answer), then `mtcars` (correct answer) below:
+
+```{r mtcars, exercise = TRUE}
 
 ```
 
-```{r grade_error-solution}
-stop("boom")
+```{r mtcars-solution}
+mtcars
 ```
 
-```{r grade_error-code-check}
-grade_code("Nice error!")
+```{r mtcars-check}
+grade_result(
+  fail_if(~identical(.result, cars), "This is the cars (not mtcars) dataset."),
+  pass_if(~identical(.result, mtcars))
+)
 ```
 
 ### 
 
 And here is the code behind the example.
 
 ````markdown
-`r ''````{r grade_code, exercise = TRUE}
-sqrt(exp(3))
+`r ''````{r mtcars, exercise = TRUE}
+
 ```
 
-`r ''````{r grade_code-solution}
-sqrt(log(2))
+`r ''````{r mtcars-solution}
+mtcars
 ```
 
-`r ''````{r grade_code-check}
-grade_code("Good job. Don't worry, things will soon get harder.")
+`r ''````{r mtcars-check}
+grade_result(
+  fail_if(~identical(.result, cars), "This is the cars (not mtcars) dataset."),
+  pass_if(~identical(.result, mtcars))
+)
+```
+````
+
+### 
+
+In the case that you wish to provide custom checking logic for errors, you can also provide a `-error-check` chunk. 
```

Comment:
Is it correct to add this explanation? If so, we should add it

```suggestion
In the case that you wish to provide custom checking logic for errors, you can also provide a `-error-check` chunk. If the student submission generates an error, gradethis will run the checking code in the `-error-check` chunk instead of the checking code in the `-check` chunk. gradethis will only run the `-error-check` code if the student submission generates an error. 
```

Reply:
Yes, good idea!